ndk/src/main/jni/metadata.c

```diff
@@ -145,6 +145,25 @@ int bsg_get_map_value_int(JNIEnv *env, bsg_jni_cache *jni_cache, jobject map,
   return 0;
 }
 
+int bsg_populate_api_level_from_map(JNIEnv *env, bsg_jni_cache *jni_cache, jobject device_data) {
+    int value = -1;
+    jstring map_key = (*env)->NewStringUTF(env, "runtimeVersions");
```

Comment:
can you confirm that `jstring`s returned by `NewStringUTF()` don't need to have `DeleteLocalRef()` called on them?

Reply:
Yes - these values will be automatically freed when the native method finishes execution and returns to the Java layer: https://docs.oracle.com/javase/7/docs/technotes/guides/jni/spec/functions.html#global_local